internal/pkg/manifests/store/builder.go

```diff
@@ -333,6 +336,10 @@ func newService(opts Options, selectorLabels, objectMetaLabels map[string]string
 		},
 	}
 
+	if opts.ServiceMonitorConfig.Enabled {
+		objectMetaLabels["thanos-self-monitoring"] = opts.Name
```

Comment:
I think this label would break the selector as we noticed with @philipgough

Reply:
correct, id be ok with merging this and follow up with a global fix for the same if that works too.